### ISSUE TITLE
Manual slice fill false positive/suggestion makes it fail

### ISSUE THREAD

**Issue by kayagokalp** [rust-lang/rust-clippy#14685]
### Summary

I recently moved to rust 1.86, and got a false positive/invalid suggestion from clippy about manual slice filling below you can find the reproducer. 

I think the report is false positive as I am not actually writing the same value into the slice for each element so I am not filling it but rather applying the same operator to each element so if we are going to be transforming it to something it should be: `bytes.iter_mut().for_each(|b| *b = !*b);`

### Reproducer

for the following piece:

```rust
for b in &mut bytes {
    *b = !*b;
}
```

I got:

```sh
warning: manually filling a slice
   --> sway-types/src/u256.rs:140:9
    |
140 | /         for b in &mut bytes {
141 | |             *b = !*b;
142 | |         }
    | |_________^ help: try: `bytes.fill(!*b);`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#manual_slice_fill
    = note: `#[warn(clippy::manual_slice_fill)]` on by default
```

If this suggestion is applied we get a compile error as `b` is not a valid variable name anymore.

I expected to see this happen: either no report or suggest `bytes.iter_mut().for_each(|b| *b = !*b);`

### Version

```text
rustc 1.86.0 (05f9846f8 2025-03-31)
binary: rustc
commit-hash: 05f9846f893b09a1be1fc8560e33fc3c815cfecb
commit-date: 2025-03-31
host: aarch64-apple-darwin
release: 1.86.0
LLVM version: 19.1.7
```

**Comment by sminez** [comment]
I'm also seeing the `manual_slice_fill` lint firing incorrectly but in my case it is suggesting calling `fill` on a custom struct that implements `IndexMut` rather than a slice (which is obviously completely wrong).

https://github.com/sminez/ad/pull/112/files#diff-b630548d20538fc5ac7df164c7c982cff9665871c95d9aab5d13889d644ab0dbR1044

```
warning: manually filling a slice
    --> src/ziplist.rs:1047:9
     |
1047 | /         for i in 0..zl.len() {
1048 | |             zl[i] = 6;
1049 | |         }
     | |_________^ help: try: `zl.fill(6);`
```

**Comment by sminez** [comment]
Thanks for the fix @profetia !
